be/src/exprs/vectorized/cast_expr.cpp

```diff
@@ -80,6 +83,98 @@ DEFINE_UNARY_FN_WITH_IMPL(TimeToNumber, value) {
+        if (real_value.ok()) {
+            builder.append(real_value.value());
+        } else {
+            builder.append_null();
+        }
+    }
+
+    return builder.build(column->is_constant());
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+static ColumnPtr cast_to_json_fn(ColumnPtr& column, std::function<JsonValue(const Datum&)> get_value) {
+    ColumnViewer<FromType> viewer(column);
+    ColumnBuilder<TYPE_JSON> builder(viewer.size());
+
+    for (int row = 0; row < viewer.size(); ++row) {
+        if (viewer.is_null(row)) {
+            builder.append_null();
+            continue;
+        }
+
+        auto real_value = get_value(viewer.value(row));
+        if (!real_value.is_null()) {
+            builder.append(&real_value);
+        } else {
+            builder.append_null();
+        }
+    }
+
+    return builder.build(column->is_constant());
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+ColumnPtr cast_json_from_bool_fn(ColumnPtr& column) {
+    return cast_to_json_fn<TYPE_BOOLEAN, TYPE_JSON>(
+            column, [](const Datum& value) { return JsonValue::from_bool((bool)value.get_int8()); });
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+ColumnPtr cast_json_from_int_fn(ColumnPtr& column) {
+    return cast_to_json_fn<FromType, TYPE_JSON>(column, [](const Datum& value) {
+        using cpp_type = RunTimeCppType<FromType>;
+        return JsonValue::from_int((int64_t)value.get<cpp_type>());
+    });
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+ColumnPtr cast_json_from_float_fn(ColumnPtr& column) {
+    return cast_to_json_fn<FromType, TYPE_JSON>(column, [](const Datum& value) {
+        using cpp_type = RunTimeCppType<FromType>;
+        return JsonValue::from_double((double)value.get<cpp_type>());
+    });
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+ColumnPtr cast_json_from_string_fn(ColumnPtr& column) {
+    return cast_to_json_fn<FromType, TYPE_JSON>(
+            column, [](const Datum& value) { return JsonValue::from_string(value.get_slice()); });
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+ColumnPtr cast_bool_from_json_fn(ColumnPtr& column) {
+    return cast_from_json_fn<FromType, ToType>(column, [](auto json) { return json->get_bool(); });
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+ColumnPtr cast_int_from_json_fn(ColumnPtr& column) {
+    return cast_from_json_fn<FromType, ToType>(column, [](auto json) { return json->get_int(); });
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+ColumnPtr cast_float_from_json_fn(ColumnPtr& column) {
+    return cast_from_json_fn<FromType, ToType>(column, [](auto json) { return json->get_double(); });
+}
+
+template <PrimitiveType FromType, PrimitiveType ToType>
+ColumnPtr cast_string_from_json_fn(ColumnPtr& column) {
+    return cast_from_json_fn<FromType, ToType>(column, [](auto json) { return json->get_string(); });
```

Comment:
cast_{float, int, ....}_{from, to}_json are tedious, suggest to use a macro or template to eliminate boilerplate code. referencing to cast_expr.ecpp

Reply:
done